x-pack/plugins/rule_registry/server/utils/create_lifecycle_executor.ts

```diff
@@ -332,8 +333,8 @@ export const createLifecycleExecutor =
     const trackedEventsToIndex = makeEventsDataMapFor(trackedAlertIds);
     const newEventsToIndex = makeEventsDataMapFor(newAlertIds);
     const trackedRecoveredEventsToIndex = makeEventsDataMapFor(trackedAlertRecoveredIds);
-    const allEventsToIndex = [
-      ...getAlertsForNotification(flappingSettings, trackedEventsToIndex),
+    const allEventsToNotify = [
+      ...getAlertsForNotification(flappingSettings, trackedEventsToIndex, notifyWhen),
```

Comment:
This doesn't seem right.  Seems like this will not write alerts is the alert would not have generated a notification (scheduled an action);  I think we still want the alert written, we just don't want the action to be scheduled.

Haven't looked at `x-pack/plugins/rule_registry/server/utils/get_alerts_for_notification.ts` thoroughly, but I'm guessing we DO want that, we just don't the filtering on the `notifyWhen` for this part of the processing, in this module.

Or maybe this is needed for some other reason I don't yet understand ...

Reply:
Oh okay, I will fix this!

Reply:
Resolved in this commit https://github.com/elastic/kibana/pull/151066/commits/f403fdcd5acc01054dfa062a79dc5667278085ae
This is my bad for misunderstanding what actually is happening in` x-pack/plugins/rule_registry/server/utils/create_lifecycle_executor.ts `.